pkgs/development/tools/build-managers/cmake/default.nix

```diff
@@ -16,12 +16,15 @@ stdenv.mkDerivation rec {
           + lib.optionalString isBootstrap "-boot"
           + lib.optionalString useNcurses "-cursesUI"
           + lib.optionalString withQt5 "-qt5UI";
-  version = "3.19.7";
+  version = "3.21.0";
 
   src = fetchurl {
     url = "${meta.homepage}files/v${lib.versions.majorMinor version}/cmake-${version}.tar.gz";
     # compare with https://cmake.org/files/v${lib.versions.majorMinor version}/cmake-${version}-SHA-256.txt
-    sha256 = "sha256-WKFfDVagr8zDzFNxI0/Oc/zGyPnb13XYmOUQuDF1WI4=";
+    # Upstream posts a hex encoded sha256, but to compare it with the output of
+    # Nix 2.4 you need to convert it to base64 format with
+    # `echo <hex> | xxd -r -p | base64`
```

Comment:
```suggestion
```
We don't need to comment this because it is common knowledge and you can also just use ``nix hash to-base64 <...>``.

Reply:
I had to figure this out so here is one data point for not *common knowledge*.

I updated the comment with instructions for `nix hash` instead, which is also not discoverable enough to omit the comment from my point of view.

Reply:
It is the first result when you google it https://www.google.com/search?q=nix+convert+hash

pretty easy to find IMO.

Reply:
Anecdotally, I didn't know about this either. I think the instruction here is useful, as the comment immediately above instructs the reader to compare it with an upstream hash.

(though I'm not really sure what the benefit of comparing the hash is; if the download is tampered with, the hash is served from the same site and would presumably be tampered with as well, and the download is done over https and declares its Content-Length so there's no risk of truncation [which would cause the file to fail to unpack anyway])

Reply:
> though I'm not really sure what the benefit of comparing the hash is; if the download is tampered with, the hash is served from the same site and would presumably be tampered with as well, and the download is done over https and declares its Content-Length so there's no risk of truncation [which would cause the file to fail to unpack anyway]

You're making a good point here, we don't seem to get any extra safety from comparing to the upstream hash. I guess I'll delete the entire comment. At least I learned about `nix(-)hash`.